### PR TITLE
Adjust peppy to handle different read types for the same sample

### PR DESCRIPTION
**Pull request by rafalstepien**
- Add handler for `AttributeError` raised by `getattr()` and return empty string if there is no value for given attribute
- Add test

### CODE REVIEW

**Comment by nsheff** [comment]
I think you will need to lint this, right?

File: peppy/project.py [hunk 1]
```diff
@@ -674,6 +660,43 @@ def _get_duplicated_and_not_duplicated_samples(
...
+            for sample in duplicated_samples:
+                attribute_value_for_sample = Project.safe_getattr(sample, attr)
+                attribute_values.append(attribute_value_for_sample)
+
+            merged_attributes[attr] = list(flatten(attribute_values))
+
+        return merged_attributes
+
+    @staticmethod
+    def safe_getattr(object: object, attribute: str):
```
Comment (nsheff) [inline peppy/project.py:694]: yes, I like this!

**Comment by rafalstepien** [comment]
> I think you will need to lint this, right?

I think I already did

**Review by nsheff — APPROVED** [review]
I didn't try running it, but the code changes look good to me!

File: peppy/project.py [hunk 2]
```diff
@@ -594,56 +594,42 @@ def _auto_merge_duplicated_names(self):
...
-                    disable=not self.is_sample_table_large,
-                )
-                if sample_names_list.count(x) > 1
-            ]
-        )
-        if not dups_set:
-            # all sample names are unique
+        duplicated_sample_ids = self._get_duplicated_sample_ids(sample_names_list)
+
+        if not duplicated_sample_ids:
```
Comment (nleroy917) [inline peppy/project.py:599]: Maybe, `if len(duplicated_sample_ids) == 0:` ?

Small thing, just slightly better readability.

File: peppy/project.py [hunk 3]
```diff
@@ -1222,6 +1245,27 @@ def _read_sample_data(self):
...
+                "index_col": False,
+                "keep_default_na": False,
+                "na_values": [""],
+            }
+            try:
+                return pd.read_csv(pth, sep=infer_delimiter(pth), **csv_kwargs)
+            except Exception as e:
+                raise SampleTableFileException(
+                    f"Could not read table: {pth}. "
+                    f"Caught exception: {getattr(e, 'message', repr(e))}"
```
Comment (nleroy917) [inline peppy/project.py:1266]: Is this a typo? Why 3  arguments to `getattr`?

Comment (nleroy917) [reply peppy/project.py:1266]: Oh. Is it a default value if the attribute doesn't exist?

File: peppy/project.py [hunk 4]
```diff
@@ -674,6 +660,43 @@ def _get_duplicated_and_not_duplicated_samples(
...
+                attribute_value_for_sample = Project.safe_getattr(sample, attr)
+                attribute_values.append(attribute_value_for_sample)
+
+            merged_attributes[attr] = list(flatten(attribute_values))
+
+        return merged_attributes
+
+    @staticmethod
+    def safe_getattr(object: object, attribute: str):
+        try:
```
Comment (nleroy917) [inline peppy/project.py:695]: Is this code not equivalent to `getattr(object, attribute, "")`? It might not be, just curious of the differences.

**Review by nleroy917 — APPROVED** [review]
Looks good! I just had questions for my own learning...

File: peppy/project.py (hunk 4 above)
Comment (rafalstepien) [reply peppy/project.py:695]: Oh it is. I didn't know that getattr has default value!

File: peppy/project.py (hunk 3 above)
Comment (rafalstepien) [reply peppy/project.py:1266]: This were changes proposed by linter so I haven't seen that

Comment (rafalstepien) [reply peppy/project.py:1266]: But yes, it is a default value

File: peppy/project.py (hunk 2 above)
Comment (rafalstepien) [reply peppy/project.py:599]: According to PEP8 
image

Comment (rafalstepien) [reply peppy/project.py:599]: https://peps.python.org/pep-0008/

**Pull request merged**